### PR TITLE
Add Dawn and Horizon cards

### PR DESCRIPTION
**Pull request by Supremicus**
Add Dawn and Horizon cards

### CODE REVIEW

**Comment by CollinHeist** [comment]
@Supremicus Awesome work! These both look exceptional. I'll start taking a look when I have the time 👍🏻

File: Supremicus/DawnTitleCard.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+    __OVERLAY_PLAY_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_play_bezel.png'))
+    __OVERLAY_REWIND = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind.png'))
+    __OVERLAY_REWIND_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind_bezel.png'))
+
+    """Source path for the gradient image"""
+    __GRADIENT_IMAGE = str(RemoteFile('Supremicus', 'ref/overlays/gradient.png'))
+
+    __slots__ = (
+        'source_file', 'output_file', 'title_text', 'season_text',
+        'episode_prefix', 'episode_text', 'hide_season_text', 'hide_episode_text',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
        'episode_text', 'hide_season_text', 'hide_episode_text',
```

File: Supremicus/DawnTitleCard.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    """Characteristics of the default title font"""
+    TITLE_FONT = str(RemoteFile('Supremicus', 'ref/fonts/HelveticaNeue-Bold.ttf'))
+    TITLE_COLOR = 'white'
+    FONT_REPLACEMENTS = {}
+
+    """Whether this CardType uses season titles for archival purposes"""
+    USES_SEASON_TITLE = True
+
+    """Standard class has standard archive name"""
+    ARCHIVE_NAME = 'dawn'
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
    ARCHIVE_NAME = 'Dawn'
```
This is just to better align with the archive naming convention of all other card types

File: Supremicus/DawnTitleCard.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+    """Source path for CRT overlays to be overlayed if enabled"""
+    __OVERLAY_PLAIN = str(RemoteFile('Supremicus', 'ref/overlays/overlay_plain.png'))
+    __OVERLAY_PLAIN_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_plain_bezel.png'))
+    __OVERLAY_PLAY = str(RemoteFile('Supremicus', 'ref/overlays/overlay_play.png'))
+    __OVERLAY_PLAY_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_play_bezel.png'))
+    __OVERLAY_REWIND = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind.png'))
+    __OVERLAY_REWIND_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind_bezel.png'))
+
+    """Source path for the gradient image"""
+    __GRADIENT_IMAGE = str(RemoteFile('Supremicus', 'ref/overlays/gradient.png'))
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
    EPISODE_TEXT_FONT = RemoteFile('Supremicus', 'ref/fonts/ExoSoft-Medium.ttf')

    """Source path for CRT overlays to be overlayed if enabled"""
    __OVERLAY_PLAIN = RemoteFile('Supremicus', 'ref/overlays/overlay_plain.png')
    __OVERLAY_PLAIN_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_plain_bezel.png')
    __OVERLAY_PLAY = RemoteFile('Supremicus', 'ref/overlays/overlay_play.png')
    __OVERLAY_PLAY_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_play_bezel.png')
    __OVERLAY_REWIND = RemoteFile('Supremicus', 'ref/overlays/overlay_rewind.png')
    __OVERLAY_REWIND_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_rewind_bezel.png')

    """Source path for the gradient image"""
    __GRADIENT_IMAGE = RemoteFile('Supremicus', 'ref/overlays/gradient.png')
```
This is so you can treat these as `Path` objects and call `resolve()` on them

File: Supremicus/DawnTitleCard.py [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+        example='https://raw.githubusercontent.com/Supremicus/tcm-images/main/Preview%20Cards/DawnTitleCard.preview.jpg',
+        creators=['Supremicus'],
+        source='remote',
+        supports_custom_fonts=True,
+        supports_custom_seasons=True,
+        supported_extras=[
+            Extra(
+                name='Episode Text Vertical Shift',
+                identifier='episode_text_vertical_shift',
+                description='Vertical Shift for Episode Text.',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='Vertical Shift for Episode Text',
```

File: Supremicus/DawnTitleCard.py [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
+                tooltip=(
+                    'Additional vertical shift to apply to the season and episode text. '
+                    'Default is <v>0</v>.<br> If multi-line issues, problem fonts may'
+                    'be fixed by <v>Fix vertical metrics</v> at <v>https://transfonter.org/</v>'
+                ),
+            ),
+            Extra(
+                name='Title Text Horizontal Shift',
+                identifier='title_text_horizontal_shift',
+                description='Horizontal shift for Title Text.',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='Horizontal shift for the title text',
```

File: Supremicus/DawnTitleCard.py [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+                tooltip=(
+                    'Horizontal shift to apply to title text to align with episode text. '
+                    'Default is <v>0</v>.'
+                ),
+            ),
+            Extra(
+                name='Stroke Text Color',
+                identifier='stroke_color',
+                description='Color to use for the episode & title text stroke.',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='Color to use for the episode & title text stroke',
```

File: Supremicus/DawnTitleCard.py [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+                tooltip='Default is <c>black</c>.'
+            ),
+            Extra(
+                name='Separator Character',
+                identifier='separator',
+                description='Character to separate season and episode text.',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='Character to separate season and episode text',
```

File: Supremicus/DawnTitleCard.py [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+                tooltip='Default is <v>•</v>.'
+            ),
+            Extra(
+                name='Horizontal Alignment',
+                identifier='h_align',
+                description='Horizontal alignment of text.',
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='Horizontal alignment of text',
```

File: Supremicus/DawnTitleCard.py [hunk 9]
```diff
@@ -0,0 +1,545 @@
...
+                tooltip='Either <v>left</v>, <v>center</v> or <v>right</v>. Default is <v>left</v>.'
+            ),
+            Extra(
+                name='CRT TV Overlay',
+                identifier='crt_overlay',
+                description='Enable CRT TV overlay.',
+                tooltip=(
+                    'Enable CRT TV overlay: <v>nobezel</v> or <v>bezel</v>.<br>'
+                    'Default is <v>None</v>.'
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='CRT TV Overlay Toggle',
                tooltip=(
                    'Either <v>nobezel</v> or <v>bezel</v>. Default is <v>None</v>.'
```

File: Supremicus/DawnTitleCard.py [hunk 10]
```diff
@@ -0,0 +1,545 @@
...
+                ),
+            ),
+            Extra(
+                name='CRT TV Watched/Unwatched Overlay',
+                identifier='crt_state_overlay',
+                description='Enable CRT TV Watched/Unwatched overlay.',
+                tooltip=(
+                    'Enable CRT TV Watched/Unwatched overlay: <v>True</v> / <v>False</v>. Default is <v>False</v>.<br>'
+                    'Will only work if CRT TV Overlay is enabled.'
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
                description='CRT TV Overlay Watched-Status Toggle',
                tooltip=(
                    'Whether to change the CRT overlay with the watched status of the Episode. '
                    'Either <v>True</v> or <v>False</v>. Default is <v>False</v>. '
                    'Will only work if the CRT TV Overlay Toggle is enabled.'
```

File: Supremicus/DawnTitleCard.py [hunk 11]
```diff
@@ -0,0 +1,545 @@
...
+        ]
+    )
+
+    class CardModel(BaseCardTypeCustomFontAllText):
+        episode_text_vertical_shift: int = 0
+        title_text_horizontal_shift: int = 0
+        stroke_color: str = 'black'
+        separator: str = '•'
+        h_align: Literal['left', 'center', 'right'] = 'left'
+        symbol: str = None
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
```
This does not seem to be used

File: Supremicus/DawnTitleCard.py [hunk 12]
```diff
@@ -0,0 +1,545 @@
...
+            gravity = 'south'
+        else:
+            gravity = 'southeast'
+
+        # Font customizations
+        font_size = 124 * self.font_size
+        interline_spacing = -20 + self.font_interline_spacing
+        interword_spacing = 50 + self.font_interword_spacing
+        kerning = -1.25 * self.font_kerning
+        vertical_shift = 50 + self.font_vertical_shift
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
```
Unused

File: Supremicus/DawnTitleCard.py [hunk 13]
```diff
@@ -0,0 +1,545 @@
...
+
+    @property
+    def add_crt_overlay_commands(self) -> ImageMagickCommands:
+        """Add the static gradient to this object's source image."""
+
+        if self.crt_overlay is None:
+            return []
+        # Select CRT overlay based on watch status
+        if self.crt_overlay == 'nobezel':
+            if self.crt_state_overlay:
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
            if self.crt_state_overlay and not self.watched:
```

File: Supremicus/DawnTitleCard.py [hunk 14]
```diff
@@ -0,0 +1,545 @@
...
+                crt_overlay_image = self.__OVERLAY_PLAY
+            elif self.crt_state_overlay and self.watched:
+                crt_overlay_image = self.__OVERLAY_REWIND
+            else:
+                crt_overlay_image = self.__OVERLAY_PLAIN
+        elif self.crt_overlay == 'bezel':
+            if self.crt_state_overlay:
```
Comment (CollinHeist) [inline Supremicus/DawnTitleCard.py]:
```suggestion
            if self.crt_state_overlay and not self.watched:
```

File: Supremicus/HorizonTitleCard.py [hunk 15]
```diff
@@ -0,0 +1,636 @@
...
+    """Source path for CRT overlays to be overlayed if enabled"""
+    __OVERLAY_PLAIN = str(RemoteFile('Supremicus', 'ref/overlays/overlay_plain.png'))
+    __OVERLAY_PLAIN_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_plain_bezel.png'))
+    __OVERLAY_PLAY = str(RemoteFile('Supremicus', 'ref/overlays/overlay_play.png'))
+    __OVERLAY_PLAY_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_play_bezel.png'))
+    __OVERLAY_REWIND = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind.png'))
+    __OVERLAY_REWIND_BEZEL = str(RemoteFile('Supremicus', 'ref/overlays/overlay_rewind_bezel.png'))
+
+    """Source path for the gradient image"""
+    __GRADIENT_IMAGE = str(RemoteFile('Supremicus', 'ref/overlays/radial_gradient.png'))
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
    EPISODE_TEXT_FONT = RemoteFile('Supremicus', 'ref/fonts/ExoSoft-Medium.ttf')

    """Source path for symbol images to be overlayed behind text"""
    __SYMBOL_IMAGE_ACOLYTE = RemoteFile('Supremicus', 'ref/symbols/acolyte.png')
    __SYMBOL_IMAGE_AHSOKA = RemoteFile('Supremicus', 'ref/symbols/ahsoka.png')
    __SYMBOL_IMAGE_ANDOR = RemoteFile('Supremicus', 'ref/symbols/andor.png')
    __SYMBOL_IMAGE_BOBAFETT = RemoteFile('Supremicus', 'ref/symbols/bobafett.png')
    __SYMBOL_IMAGE_MANDALORIAN = RemoteFile('Supremicus', 'ref/symbols/mandalorian.png')
    __SYMBOL_IMAGE_OBIWAN = RemoteFile('Supremicus', 'ref/symbols/obiwan.png')
    __SYMBOL_IMAGE_WITCHER = RemoteFile('Supremicus', 'ref/symbols/witcher.png')

    """Alignment overlay image"""
    __ALIGNMENT_OVERLAY_IMAGE = RemoteFile('Supremicus', 'ref/overlays/overlay_alignment.png')

    """Source path for CRT overlays to be overlayed if enabled"""
    __OVERLAY_PLAIN = RemoteFile('Supremicus', 'ref/overlays/overlay_plain.png')
    __OVERLAY_PLAIN_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_plain_bezel.png')
    __OVERLAY_PLAY = RemoteFile('Supremicus', 'ref/overlays/overlay_play.png')
    __OVERLAY_PLAY_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_play_bezel.png')
    __OVERLAY_REWIND = RemoteFile('Supremicus', 'ref/overlays/overlay_rewind.png')
    __OVERLAY_REWIND_BEZEL = RemoteFile('Supremicus', 'ref/overlays/overlay_rewind_bezel.png')

    """Source path for the gradient image"""
    __GRADIENT_IMAGE = RemoteFile('Supremicus', 'ref/overlays/radial_gradient.png')
```

File: Supremicus/HorizonTitleCard.py [hunk 16]
```diff
@@ -0,0 +1,636 @@
...
+            "for shows like Stranger Things with a retro theme.",
+        ]
+    )
+
+    class CardModel(BaseCardTypeCustomFontAllText):
+        episode_text_vertical_shift: int = 0
+        stroke_color: str = 'black'
+        separator: str = '•'
+        h_align: Literal['left', 'right'] = 'left'
+        symbol: str = None
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
        symbol: Optional[str] = None
```

File: Supremicus/HorizonTitleCard.py [hunk 17]
```diff
@@ -0,0 +1,636 @@
...
+                tooltip=(
+                    'Additional vertical shift to apply to the season and episode text. '
+                    'Default is <v>0</v>.<br> If multi-line issues, problem fonts may'
+                    'be fixed by <v>Fix vertical metrics</v> at <v>https://transfonter.org/</v>'
+                ),
+            ),
+            Extra(
+                name='Stroke Text Color',
+                identifier='stroke_color',
+                description='Color to use for the episode & title text stroke.',
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='Color to use for the episode & title text stroke',
```

File: Supremicus/HorizonTitleCard.py [hunk 18]
```diff
@@ -0,0 +1,636 @@
...
+                tooltip='Default is <c>black</c>.'
+            ),
+            Extra(
+                name='Separator Character',
+                identifier='separator',
+                description='Character to separate season and episode text.',
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='Character to separate season and episode text',
```

File: Supremicus/HorizonTitleCard.py [hunk 19]
```diff
@@ -0,0 +1,636 @@
...
+                tooltip='Default is <v>•</v>.'
+            ),
+            Extra(
+                name='Horizontal Alignment',
+                identifier='h_align',
+                description='Horizontal alignment of text and symbol.',
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='Horizontal alignment of the text and symbol',
```

File: Supremicus/HorizonTitleCard.py [hunk 20]
```diff
@@ -0,0 +1,636 @@
...
+                tooltip='Either <v>left</v> or <v>right</v>. Default is <v>left</v>.'
+            ),
+            Extra(
+                name='Symbol',
+                identifier='symbol',
+                description='Add a custom symbol.',
+                tooltip=(
+                    '<b>Built-in symbols</b>: <v>acolyte</v>, <v>ahsoka</v>, <v>andor</v>,'
+                    ' <v>bobafett</v>, <v>mandalorian</v>, <v>obiwan</v>, <v>witcher</v>.<br>'
+                    '<b>Custom</b>: <v>logo</v>. Files > Logo > Upload. '
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='Add a custom symbol behind the text',
                tooltip=(
                    'Either <v>acolyte</v>, <v>ahsoka</v>, <v>andor</v>, <v>bobafett<v>, '
                    '<v>mandalorian</v>, <v>obiwan</v>, or <v>witcher</v> to use a built-in '
                    'symbol, or <v>logo</v> to use the Series logo.'
```

File: Supremicus/HorizonTitleCard.py [hunk 21]
```diff
@@ -0,0 +1,636 @@
...
+                ),
+            ),
+            Extra(
+                name='CRT TV Overlay',
+                identifier='crt_overlay',
+                description='Enable CRT TV overlay.',
+                tooltip=(
+                    'Enable CRT TV overlay: <v>nobezel</v> or <v>bezel</v>.<br>'
+                    'Default is <v>None</v>.'
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='CRT TV Overlay Toggle',
                tooltip=(
                    'Either <v>nobezel</v> or <v>bezel</v>. Default is <v>None</v>.'
```

File: Supremicus/HorizonTitleCard.py [hunk 22]
```diff
@@ -0,0 +1,636 @@
...
+                ),
+            ),
+            Extra(
+                name='CRT TV Watched/Unwatched Overlay',
+                identifier='crt_state_overlay',
+                description='Enable CRT TV Watched/Unwatched overlay.',
+                tooltip=(
+                    'Enable CRT TV Watched/Unwatched overlay: <v>True</v> / <v>False</v>. Default is <v>False</v>.<br>'
+                    'Will only work if CRT TV Overlay is enabled.'
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='CRT TV Overlay Watched-Status Toggle',
                tooltip=(
                    'Whether to change the CRT overlay with the watched status of the Episode. '
                    'Either <v>True</v> or <v>False</v>. Default is <v>False</v>. '
                    'Will only work if the CRT TV Overlay Toggle is enabled.'
```

File: Supremicus/HorizonTitleCard.py [hunk 23]
```diff
@@ -0,0 +1,636 @@
...
+                ),
+            ),
+            Extra(
+                name='Alignment Overlay',
+                identifier='alignment_overlay',
+                description='Enable Alignment Overlay.',
+                tooltip=(
+                    'Enables alignment overlay to help assist adjusting offsets for'
+                    ' misaligned custom fonts. Guides per 10 pixles. <v>True</v> / <v>False</v>.' 
+                    ' Default is <v>False</v>.'
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
                description='Alignment Overlay Toggle',
                tooltip=(
                    'Enable an alignment overlay to help assist adjusting offsets for '
                    'misaligned custom fonts. The overlay has guiding lines every 10 pixels '.
                    'Either <v>True</v> or <v>False</v>. Default is <v>False</v>.'
```

File: Supremicus/HorizonTitleCard.py [hunk 24]
```diff
@@ -0,0 +1,636 @@
...
+    """Characteristics of the default title font"""
+    TITLE_FONT = str(RemoteFile('Supremicus', 'ref/fonts/HelveticaNeue-Bold.ttf'))
+    TITLE_COLOR = 'white'
+    FONT_REPLACEMENTS = {}
+
+    """Whether this CardType uses season titles for archival purposes"""
+    USES_SEASON_TITLE = True
+
+    """Standard class has standard archive name"""
+    ARCHIVE_NAME = 'horizon'
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
    ARCHIVE_NAME = 'Horizon'
```

File: Supremicus/HorizonTitleCard.py [hunk 25]
```diff
@@ -0,0 +1,636 @@
...
+            x = -700
+        else:
+            x = 700
+
+        # Font customizations
+        font_size = 124 * self.font_size
+        interline_spacing = -26 + self.font_interline_spacing
+        interword_spacing = 50 + self.font_interword_spacing
+        kerning = -1.25 * self.font_kerning
+        vertical_shift = 40 + self.font_vertical_shift
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
```
Unused

File: Supremicus/HorizonTitleCard.py [hunk 26]
```diff
@@ -0,0 +1,636 @@
...
+
+    @property
+    def add_crt_overlay_commands(self) -> ImageMagickCommands:
+        """Add the static gradient to this object's source image."""
+
+        if self.crt_overlay is None:
+            return []
+        # Select CRT overlay based on watch status
+        if self.crt_overlay == 'nobezel':
+            if self.crt_state_overlay:
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
            if self.crt_state_overlay and not self.watched:
```

File: Supremicus/HorizonTitleCard.py [hunk 27]
```diff
@@ -0,0 +1,636 @@
...
+                crt_overlay_image = self.__OVERLAY_PLAY
+            elif self.crt_state_overlay and self.watched:
+                crt_overlay_image = self.__OVERLAY_REWIND
+            else:
+                crt_overlay_image = self.__OVERLAY_PLAIN
+        elif self.crt_overlay == 'bezel':
+            if self.crt_state_overlay:
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion
            if self.crt_state_overlay and not self.watched:
```

File: Supremicus/HorizonTitleCard.py [hunk 28]
```diff
@@ -0,0 +1,636 @@
...
+        Add alignment overlay image.
+        """
+        alignment_image = self.__ALIGNMENT_OVERLAY_IMAGE
+
+        if self.alignment_overlay is False:
+            return []
+
+        return [
+            f'"{alignment_image.resolve()}"',
+            f'-composite',
```
Comment (CollinHeist) [inline Supremicus/HorizonTitleCard.py]:
```suggestion

        if not self.alignment_overlay:
            return []

        return [
            f'"{self.__ALIGNMENT_OVERLAY_IMAGE.resolve()}"',
            f'-composite',
```

**Review by CollinHeist — CHANGES_REQUESTED** [review]
Looks really good! If you are able to make these changes (at least for the ones which are non-formatting related) then it should be good.

**Comment by Supremicus** [comment]
@CollinHeist Cards updated from code review suggestions

**Comment by CollinHeist** [comment]
Looks great! Thanks again.

**Pull request merged**